### PR TITLE
OBPIH-3932 Fix creating new source on edit order item modal

### PR DESCRIPTION
**Pull request by DominikaHatala**

### CODE REVIEW

**Review by awalkowiak — COMMENTED** [review]
How does this fix the issue?

File: grails-app/controllers/org/pih/warehouse/order/OrderController.groovy
Comment (awalkowiak) [inline grails-app/controllers/org/pih/warehouse/order/OrderController.groovy:670]: You are basically creating the productSupplier for the second time here.

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: grails-app/controllers/org/pih/warehouse/order/OrderController.groovy
Comment (DominikaHatala) [reply grails-app/controllers/org/pih/warehouse/order/OrderController.groovy:670]: In this case you can create without having enter any data for new product supplier or without having product supplier chosen before. In the previous if you can only get already available product supplier or create new with given data.

**Comment by DominikaHatala** [comment]
@awalkowiak Please take a look at my comment in requested changes.

File: grails-app/controllers/org/pih/warehouse/order/OrderController.groovy
Comment (awalkowiak) [reply grails-app/controllers/org/pih/warehouse/order/OrderController.groovy:670]: 1. `productSupplierDataService.getOrCreateNew(params)` will call `productSupplierDataService.createProductSupplierWithoutPackage(params)` inside. So it's triggered twice.
2. If you will have params.productSupplier == "Create New" and params.sourceCode, then the line 666 will trigger. (first, you create it in line 660, and then line 664 will find it and will complain that it already exists).
3. Indentation of try-catch seems to be a little off.

**Comment by DominikaHatala** [comment]
@awalkowiak I've applied some changes.

**Comment by jmiranda** [comment]
@awalkowiak I'll let you merge this when you feel good about it. I don't love the magic string ("Create new") since the string should technically be translated. And even if it's a static string in the select list, it could lead to a regression if someone were to change that value in the UI without changing this string.

**Comment by awalkowiak** [comment]
@jmiranda The "Create new" is an option that is injected on the frontend, but you right. I think we could refactor this after release and instead of injecting it on the frontend, we could inject from the backend as an option with translated label and id made out of some constant.

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**
